plugins/pulp_rpm/plugins/importers/yum/sync.py

```diff
@@ -77,12 +81,49 @@ def set_progress(self):
     @property
     def sync_feed(self):
         """
-        :return:    the URL of the feed we should sync
-        :rtype:     str
+        :return:    list the URLs of the feed we should sync
+        :rtype:     list
         """
         repo_url = self.call_config.get(importer_constants.KEY_FEED)
-        if repo_url is not None and not repo_url.endswith('/'):
-            repo_url += '/'
+        if repo_url:
+            repo_url_slash = repo_url
+            self.tmp_dir = tempfile.mkdtemp(dir=self.working_dir)
+            if not repo_url.endswith('/'):
+                    repo_url_slash = repo_url + '/'
+            try:
+                self.check_metadata(repo_url_slash)
+                return [repo_url_slash]
+            except:
```

Comment:
We shouldn't catch all Exceptions here. Instead, we should catch the specific Exception that would be raised if repo_url is not a mirrorlist.